config/errors.go

```diff
@@ -13,6 +15,83 @@ func (e ErrMapNotFound) Error() string {
 	return fmt.Sprintf("config: map (%v) not found", e.MapName)
 }
 
+type ErrParamTokenReserved struct {
+	MapName   string
+	Parameter provider.QueryParameter
+}
+
+func (e ErrParamTokenReserved) Error() string {
+	return fmt.Sprintf("config: map %s has parameter %s referencing reserved token %s",
```

Comment:
I appreciate that you're trying to follow the format of `"map %s something"`, but I find these error messages difficult to understand. I'll suggest some alternatives.
```suggestion
	return fmt.Sprintf("config: map %s parameter %s uses a reserved token name %s",
```